### ISSUE TITLE
Datei ersetzen

### ISSUE THREAD

**Issue by Znrl** [contao/contao#2177]
Hallo zusammen.

**Problem:**
Eine Datei ist an mehreren Stellen auf der Website, vtl. auch in News/Veranstaltungen eingebunden (z.B: ein Flyer).
Irgendwann muss diese ersetzt werden.
Es gibt natürlich mehr oder weniger praktikable Lösungen.
Über die Datenbank, ersetzen per FTP, etwas bastelei und mit fixem Pfad einbinden, ...

Fehlt aber entsprechender Zugriff o.ä. muss man wissen oder suchen, wo diese Datei verwendet wird und jede Stelle einzeln ändern.

**Potentielles Feature:**
**"Datei ersetzen"**, wenn man eine Datei in der Dateiverwaltung bearbeitet.

Ich weiß nicht ob irgendwelche Änderungen bzgl. der Dateiverwaltung geplant sind. Z.B. hinsichtlich Versionierung (Papierkorb o.ä.), was ja auch bei allen anderen Inhaltselementen in letzter Zeit nochmal etwas im Fokus war. Das könnte mit dem hier verbunden werden.

Es wäre hier vermutl. notwendig entweder die alten auf die neuen UUIDs zu mappen oder die Verwendung von Dateien auf der Seite zu tracken (könnte auch ein nettes Feature an sich sein, wenn man sich alle Stellen anzeigen lassen könnte, an der die Datei benutzt wird, zumindest Filepicker und TinyMCE).
Gib natürlich noch ein paar mehr Fragen.
Zumindest Warnmeldungen, wenn die neue Datei eine andere Dateieindung hat. Auch Beim Ersetzen von Bildern, wenn die Auflösung eine andere ist. Oder eine generelle Blacklist für manche Dateitypen?
Das ganze ist vtl. also nicht ganz trivial umzusetzen.

Ist das ein Problem, das noch andere hier kennen?
Gibt es vielleicht eine praktikable Lösung, die ich nicht gefunden habe?

**Comment by fritzmg** [comment]
Einfach die Datei mit einer anderen Datei ersetzen, die den selben Namen hat? ;)

**Comment by Znrl** [comment]
Ja per FTP (wenn man "schneller" als die Synchronisation durch Contao ist) geht das aber im Dateimanager hat man doch das Problem, dass die Datei nicht die selbe UUID hat. Der TinyMCE und Filepicker speichert ja die UUID.
Oder nicht? Das hat ja wieder Vorteile aber eben, dachte ich bisher, den Nachteil, dass ein Ersetzen über die Dateiverwaltung nicht geht.

**Comment by fritzmg** [comment]
Nein, du kannst die Datei direkt im Dateimanager ersetzen. Die UUID ändert sich dadurch nicht soweit ich weiß.

**Comment by Znrl** [comment]
Interessant, jetzt frage ich mich warum ich so felsenfest davon überzeugt war, dass das mir die Datei nicht ersetzt sondern "dateiname-1" mäßig umbennent, dass ich schon seit je her die alten Dateien lösche und die neuen dann hochlade...
Vielleicht Windows und Wordpress (dort braucht man dazu meine ich ein Plugin) geschädigt?

Bzw. beim ersten Test hat es nicht funktioniert. Da musste ich zuerst noch mal die Synchronisation manuell anstoßen. Vielleicht liegt es etwas daran, dass ich die Dateistruktur gerne per FTP anlege und wenn man dort etwas ändert und nicht mehr synchronisiert ändert sich an der UUID beim "reupload" scheinbar u.U. doch was.

Danke, ist mir jetzt etwas peinlich ... :>

**Comment by ausi** [comment]
Eine explizite Funktion um eine Datei zu ersetzen fände ich aber schon gut. Jetzt muss ich die Datei zuerst lokal genau so benennen wie die Datei in der Dateiverwaltung heißt damit sie beim Upload ersetzt wird.

Ein Button beim editieren der Datei mit dem sie sich durch einen Upload ersetzten lässt unabhängig vom Dateinamen der lokalen Datei wäre deutlich selbsterklärender denke ich.

**Comment by Znrl** [comment]
Es hat tatsächlich ein wenig was von einem hidden feature. Da die Funktionalität grundsätzlich ja schon da ist und solange man da nicht noch was ändert, wäre es wohl auch deutich einfacher umzusetzen, als meine Überlegungen.
Nachdem ich da jetzt zig Jahre nicht drauf gekommen bin, hättest du da meine Stimme ;)

**Comment by m-vo** [comment]
> Ein Button beim editieren der Datei mit dem sie sich durch einen Upload ersetzten lässt unabhängig vom Dateinamen der lokalen Datei wäre deutlich selbsterklärender denke ich.

bzw. drauf droppen = ersetzen (ggf. mit vorherigem Dialog)

**Comment by fritzmg** [comment]
@ausi draggy droppy task für's Entwickler Treffen 😁

**Comment by chriza** [comment]
Die Extension hh-com/contao-filereplace löst die Fragestellung
